clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java

```diff
@@ -96,6 +99,121 @@ private void testInvalidInputs(String configValue) {
         }
     }
 
+    @Test
+    public void testClassConfigs() {
+        class RestrictedClassLoader extends ClassLoader {
+            public RestrictedClassLoader() {
+                super(null);
+            }
+            @Override
+            protected Class<?> findClass(String name) throws ClassNotFoundException {
+                if (name.equals(ClassTestConfig.DEFAULT_CLASS.getName()) || name.equals(ClassTestConfig.RESTRICTED_CLASS.getName()))
+                    return null;
+                else
+                    return ClassTestConfig.class.getClassLoader().loadClass(name);
+            }
+        }
+
+        ClassTestConfig testConfig;
```

Comment:
Nitpick: in Kafka, we generally set the field where it's first used unless it's an if/else where the field could be set in either.